frontend/public/components/monitoring/dashboards/graph.tsx

```diff
@@ -1,36 +1,69 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
 
+import * as UIActions from '../../../actions/ui';
 import { RootState } from '../../../redux';
 import { FormatLegendLabel, QueryBrowser } from '../query-browser';
 
-const Graph_: React.FC<Props> = ({
+type StateProps = {
+  endTime: number;
+  timespan: number;
+};
+
+type DispatchProps = {
+  setEndTime: (endTime: number) => any;
+  setTimespan: (timespan: number) => any;
+};
+
+type OwnProps = {
+  formatLegendLabel?: FormatLegendLabel;
+  isStack: boolean;
+  pollInterval: number;
+  queries: string[];
+};
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+const Graph: React.FC<Props> = ({
+  endTime,
   formatLegendLabel,
   isStack,
   pollInterval,
   queries,
+  setEndTime,
+  setTimespan,
   timespan,
-}) => (
-  <QueryBrowser
-    defaultSamples={30}
-    formatLegendLabel={formatLegendLabel}
-    hideControls
-    isStack={isStack}
-    pollInterval={pollInterval}
-    queries={queries}
-    timespan={timespan}
-  />
-);
-const Graph = connect(({ UI }: RootState) => ({
-  timespan: UI.getIn(['monitoringDashboards', 'timespan']),
-}))(Graph_);
-
-type Props = {
-  formatLegendLabel?: FormatLegendLabel;
-  isStack: boolean;
-  pollInterval: number;
-  queries: string[];
-  timespan: number;
+}) => {
+  const onZoom = React.useCallback(
+    (from, to) => {
+      setEndTime(to);
+      setTimespan(to - from);
+    },
+    [setEndTime, setTimespan],
+  );
+
+  return (
+    <QueryBrowser
+      defaultSamples={30}
+      fixedEndTime={endTime}
+      formatLegendLabel={formatLegendLabel}
+      hideControls
+      isStack={isStack}
+      onZoom={onZoom}
+      pollInterval={pollInterval}
+      queries={queries}
+      timespan={timespan}
+    />
+  );
 };
 
-export default Graph;
+export default connect<StateProps, DispatchProps, OwnProps>(
```

Comment:
Consider `useSelector` and `useDispatch` instead of `connect`.

https://react-redux.js.org/next/api/hooks

Reply:
Good point 😄 . Now switched to use `useDispatch` and `useSelector`.